src/components/[guild]/EditGuild/components/SocialLinks.tsx

```diff
@@ -33,46 +47,69 @@ const SocialLinks = (): JSX.Element => {
 
   const definedSocialLinks = useWatch({ control, name: "socialLinks" })
 
+  const validateUrl = (input: string) => {
+    const { success } = z.string().url().safeParse(input)
+    return success || "Invalid link format."
+  }
   return (
     <SimpleGrid columns={{ base: 1, md: 2 }} gap={2}>
       {Object.entries(definedSocialLinks ?? {})
         .filter(([, value]) => typeof value !== "undefined")
-        .map(([key]) => (
-          <GridItem key={key}>
-            <FormControl isInvalid={!!errors?.socialLinks?.[key]} isRequired>
-              <InputGroup size="lg">
-                <InputLeftElement>
-                  {socialLinkOptions.find((sl) => sl.value === key).img}
-                </InputLeftElement>
-                <Input
-                  type="url"
-                  {...register(`socialLinks.${key}`, {
-                    required: "This field is required.",
-                  })}
-                  placeholder={
-                    socialLinkOptions.find((sl) => sl.value === key).label
-                  }
-                />
-                <InputRightElement>
-                  <CloseButton
-                    aria-label="Remove link"
-                    size="sm"
-                    rounded="full"
-                    onClick={() =>
-                      setValue(`socialLinks.${key}`, undefined, {
-                        shouldDirty: true,
-                      })
+        .map(([key]) => {
+          const { onBlur, ...registerBindings } = register(`socialLinks.${key}`, {
+            required: "This field is required.",
+            validate: validateUrl,
+          })
+          return (
+            <GridItem key={key}>
+              <FormControl isInvalid={!!errors?.socialLinks?.[key]} isRequired>
+                <InputGroup size="lg">
+                  <InputLeftElement>
+                    {socialLinkOptions.find((sl) => sl.value === key).img}
+                  </InputLeftElement>
+                  <Input
+                    type="url"
+                    {...registerBindings}
+                    onBlur={(...args) => {
```

Comment:
It's a bit strange that we first validate the field, then we actually change its value on blur, and it becomes valid again. Could we either change it's value in the `onChange` handler or only when the user submits the form? What do you think?

Reply:
I changed it now so the auto-fill only triggers when new social link is added. I find this approach much more gentle on the user.